spec/lib/extensions/ar_visible_attribute_spec.rb

```diff
@@ -0,0 +1,47 @@
+RSpec.describe ArVisibleAttribute do
+  context ".visible_attribute_names" do
+    let(:klass) do
+      Class.new(ApplicationRecord) do
+        # ApplicationRecord already included ArVisibleAttribute
+        self.table_name = "vms"
+      end
+    end
+
+    it "shows regular attributes" do
+      expect(klass.visible_attribute_names).to include("type")
+    end
+
+    it "hides hidden attributes" do
+      klass.hide_attribute :type
+      expect(klass.visible_attribute_names).not_to include("type")
+    end
+
+    it "only hides for specified class" do
+      klass.hide_attribute :type
+      expect(Vm.visible_attribute_names).to include("type")
+    end
+  end
+
+  context ".hidden_attribute_names" do
+    let(:klass) do
+      Class.new(ApplicationRecord) do
+        # ApplicationRecord already included ArVisibleAttribute
+        self.table_name = "vms"
+      end
+    end
+
+    it "starts with no hidden attributes" do
+      expect(klass.hidden_attribute_names).to be_empty
+    end
+
+    it "hides hidden attributes" do
+      klass.hide_attribute :type
+      expect(klass.hidden_attribute_names).not_to include("type")
```

Comment:
I think this need to be `expect(klass.hidden_attribute_names).to include("type")`